packages/textmate-grammars/src/browser/tcl.ts

```diff
@@ -0,0 +1,46 @@
+/********************************************************************************
+ * Copyright (C) 2018 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { LanguageGrammarDefinitionContribution, TextmateRegistry } from '@theia/monaco/lib/browser/textmate';
+import { injectable } from 'inversify';
+
+@injectable()
+export class TclContribution implements LanguageGrammarDefinitionContribution {
+
+    readonly id = 'tcl';
+    readonly scopeName = 'source.tcl';
+
+    registerTextmateLanguage(registry: TextmateRegistry) {
+        monaco.languages.register({
+            id: this.id,
+            aliases: ['TCL', 'tcl', 'Tcl'],
+            // .exp are Expect files, which are written in TCL.
+            extensions: ['.tcl', '.exp'],
+        });
+        const grammar = require('../../data/Tcl.plist');
+        registry.registerTextmateGrammarScope(this.scopeName, {
+            async getGrammarDefinition() {
+                const grammarResponse: Response = await fetch(grammar);
```

Comment:
This works in general. Actually, I would really prefer to convert the `Tcl.plist` to a `Tcl.plist.json` once and align with other examples here. You can do this using `plutil -convert json Tcl.plist -o Tcl.plist.json` on macOS (Xcode), or by using any conversion service found on the web. 
It's the additional `GET` request with outweighs the convenience to use the original plist file.

Reply:
Well, I saw the necessity of having to GET the grammar file as an advantage here.  There are many grammars, and they tend to be somewhat big.  Since a user a likely to use only one, or two of them, I think it makes sense to have them downloaded on a per-needed basis, rather than all at the beginning, so we avoid downloading things we don't need.

We could still do that with json files though.

Reply:
Good point. I also it'S good to have an instance of plist usage. So we know it works :)